src/game/Spells/SpellAuras.cpp

```diff
@@ -9401,6 +9392,8 @@ void Aura::HandleAuraControlVehicle(bool apply, bool Real)
 
     if (apply)
     {
+        if (caster->IsBoarded() && caster->GetTransportInfo()->GetTransport()->GetObjectGuid() == target->GetObjectGuid())
```

Comment:
What is this snuck in change that looks like stacking code or scripting code.

Reply:
This is to make sure that passengers are properly unboarded when their seat is changed by spell and not by choice
It's been to long to remember why exactly I put this here, I'll add a new comment once I remember

Reply:
Ok then need to confirm things, is the previous board a spell based one, if yes, my #1 guess is that every entity can have at most one SPELL_AURA_CONTROL_VEHICLE and should be kicked off the entity during stack check step. Thats the simple case. (lets say mob has 55555 aura and 66666 gets applied)

The second case is when npc should swap seats on a vehicle by altering vehicle aura amount, this is more complex because exit movement should not be triggered on unboard. (you know the one that launches you to fuck wherever)

You can confirm if its first or second by checking sniffs and whether exit movement happens. Also need to show which part of the script triggers this piece of code.

I think I have a good idea why you had to do it and that the issue really exists but what I am trying to say is that this is not a very good solution imho without further analysis and further details.

Reply:
according to packet data no exit movement happens, just another board movement

Reply:
Alright, thats a step in a direction. Check if always on that particular aura removal unboard there is no exit movement, and if yes, we can leverage GetAuraScriptCustomizationValue overriding in spell script to tell it to merely unboard without movement. (I will code it if you confirm it, since its a 5 min job) If there are cases where same aura produces unboard and just exit, based on what caused the event, then we need to go with a route where On Unapply, aura needs to just trigger exit from seat and aura apply will trigger entry into new seat.

TC code seems to imply this is a common thing, and they specifically detect that such an unboard is in progress.

<img width="840" alt="image" src="https://github.com/cmangos/mangos-wotlk/assets/7995382/bb90ee25-4d8e-4077-8b6c-957992a67b98">

As such, need to exactly know what would trigger this code and when to move forward. Spell ids, scenarios, so its testable.

Reply:
This exact Aura never triggers exit movement, even when it drops off.
SplineFlags: 8388608
![image](https://github.com/cmangos/mangos-wotlk/assets/3718129/9d246388-dcfd-4a3d-acb5-2b9fe7420c3f)

Reply:
https://github.com/cmangos/mangos-wotlk/commit/55af7484b110d31ace3625b669731e5d0fc3b543

Tested, fixed.